src/main/java/org/testng/internal/DynamicGraph.java

```diff
@@ -46,6 +53,10 @@ public void addNode(T node) {
     m_nodesReady.add(node);
   }
 
+  public void addEdge(int weight, T from, T to, int order) {
```

Comment:
I got confused trying to understand the parameter ordering, I suggest you make both the `Edge` constructor and the `addEdge()` method use the same ordering, e.g. `from`, `to`, `weight`, `order`.